test/units/TEST-23-UNIT-FILE.ExtraFileDescriptors.sh

```diff
@@ -51,6 +51,17 @@ timeout 10s bash -xec 'while [[ "$(systemctl show -P SubState test-23-extra-fd.s
 assert_eq "$(systemctl show -P Result "$TEST_UNIT")" "success"
 assert_eq "$(systemctl show -P ExecMainStatus "$TEST_UNIT")" "0"
 
+# Verify extra file descriptors stay accessible even after service manager re-executes
+systemctl daemon-reexec
+
+systemctl restart "$TEST_UNIT"
+
+# shellcheck disable=SC2016
+timeout 10s bash -xec 'while [[ "$(systemctl show -P SubState test-23-extra-fd.service)" != "exited" ]]; do sleep .5; done'
```

Comment:
Actually the loop here can be replaced with a single assertion on SubState being exited. For oneshot services the job finishes only when all commands finish execution after all.

Reply:
Oh interesting...I didn't know this. I just ran a `sleep 100` oneshot service and indeed `systemctl restart` actually waited 100 seconds. TIL!

Reply:
Fixed in latest!